### PR TITLE
Refactor/search

### PR DESCRIPTION
**Pull request by zhouyuhe**
Be able to search by both names and symbols with the most relevant one on the top

### CODE REVIEW

File: server/components/searchQuery.js [hunk 1]
```diff
@@ -1,14 +1,24 @@
...
 
-    socket.emit("companySymbols", topTen);
+    const dataSort = filterData.sort((a, b) => {
+      const aStart =
+        a.symbol.match(new RegExp("^" + inputQuery.toUpperCase())) || [];
+      const bStart =
+        b.symbol.match(new RegExp("^" + inputQuery.toUpperCase())) || [];
+
+      if (aStart.length != bStart.length) return bStart.length - aStart.length;
+      else return a.symbol > b.symbol ? 1 : -1;
```
Comment (Sharakai) [inline server/components/searchQuery.js]: nit: You can remove the `else` word

File: server/components/searchQuery.js [hunk 2]
```diff
@@ -1,14 +1,24 @@
...
       search =>
         search.symbol.toLowerCase().indexOf(inputQuery.toLowerCase()) !== -1 ||
         search.name.toLowerCase().indexOf(inputQuery.toLowerCase()) !== -1
     );
-    const topTen = filteredData.slice(0, 10);
 
-    socket.emit("companySymbols", topTen);
+    const dataSort = filterData.sort((a, b) => {
+      const aStart =
+        a.symbol.match(new RegExp("^" + inputQuery.toUpperCase())) || [];
```
Comment (Sharakai) [inline server/components/searchQuery.js]: Perf: I'd take the `new RegExp("^" + inputQuery.toUpperCase())` out of this sort and re-use it, so it doesn't have to create a new RegExp object twice on every function invocation.

File: server/server.js [hunk 3]
```diff
@@ -1,6 +1,9 @@
 require("dotenv").config();
+// express: fast, minimailist framework for node.js
```
Comment (Sharakai) [inline server/server.js]: nit: I like that you're researching the different libraries, and making notes on them. I wouldn't leave these comments here though - people can easily google `express` if they don't know what it is.

I'd say that a list of libraries / frameworks could be in a `.md` file within the `/server` folder, describing your library/framework & design decisions

File: server/server.js [hunk 4]
```diff
@@ -1,6 +1,9 @@
...
 const express = require("express");
+//use http server and one client must require it
```
Comment (steinalex) [inline server/server.js]: Please remove these comments

File: server/components/searchQuery.js [hunk 5]
```diff
@@ -1,14 +1,24 @@
...
+
+    const filterData = symbols.filter(
       search =>
         search.symbol.toLowerCase().indexOf(inputQuery.toLowerCase()) !== -1 ||
         search.name.toLowerCase().indexOf(inputQuery.toLowerCase()) !== -1
     );
-    const topTen = filteredData.slice(0, 10);
 
-    socket.emit("companySymbols", topTen);
+    const dataSort = filterData.sort((a, b) => {
```
Comment (steinalex) [inline server/components/searchQuery.js]: Consider using a more declarative name for this function

File: server/server.js (hunk 3 above)
Comment (zhouyuhe) [reply server/server.js]: good idea

File: server/components/searchQuery.js [hunk 6]
```diff
@@ -1,14 +1,24 @@
+const regExp = input => new RegExp("^" + input.toUpperCase());
```
Comment (steinalex) [inline server/components/searchQuery.js]: nit: This could be called something more relevant, like `matchSymbol`

File: .gitignore [hunk 7]
```diff
@@ -16,6 +16,10 @@
 .env.*
 .env
 
+#vscode
+.vscode/settings.js
+.vscode/
```
Comment (steinalex) [inline .gitignore]: Adding an `*` after .vscode (e.g. `.vscode*`) will ensure any folder sub items are ignored

**Pull request merged**